File: 파이썬/배달.py
from collections import deque


def solution(N, road, K):

    nodes = {}
    dist = {i: float('inf') if i != 1 else 0 for i in range(1, N+1)}
    for v1, v2, d in road:
        nodes[v1] = nodes.get(v1, []) + [(v2, d)]
        nodes[v2] = nodes.get(v2, []) + [(v1, d)]
    que = deque([1])
    while que:
        cur_node = que.popleft()
        for nxt_node, d in nodes[cur_node]:
            if dist[nxt_node] > dist[cur_node] + d:
                dist[nxt_node] = dist[cur_node] + d
                que.append(nxt_node)

    return len([True for dist in dist.values() if dist <= K])

File: 자바스크립트/배달.js
function solution(N, road, K) {
    let [costTable, distTable] = [{}, {}];
    for (let i = 1; i <= N; i++) {
        costTable[i] = [];
        distTable[i] = i > 1 ? Infinity : 0;
    }

    for (const [start, end, dist] of road) {
        costTable[start].push([end, dist]);
        costTable[end].push([start, dist]);
    }

    const START = 1;
    let que = [START];
    while (que.length > 0) {
        const curNode = que.shift();
        for (const [nxtNode, dist] of costTable[curNode]) {
            const newDist = distTable[curNode] + dist;
            if (newDist < distTable[nxtNode]) {
                distTable[nxtNode] = newDist;
                que.push(nxtNode);
            }
        }
    }

    return Object.values(distTable).filter(x => x <= K).length;
}
